onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperatorBatchNormalization.cpp

```diff
@@ -101,15 +162,46 @@ class DmlOperatorBatchNormalization : public DmlOperator, BatchNormalizationHelp
 
 void CALLBACK QueryBatchNormalization(IMLOperatorSupportQueryContextPrivate* context, /*out*/ bool* isSupported)
 {
-    // training_mode=1 is unsupported as it isn't needed for inference (https://github.com/onnx/onnx/pull/3333).
+    *isSupported = false;
 
+    // training_mode=1 is unsupported as it isn't needed for inference (https://github.com/onnx/onnx/pull/3333).
     MLOperatorAttributes attributes(context);
     int32_t trainingMode = attributes.GetOptionalAttribute<int32_t>(AttrName::TrainingMode, 0);
-    *isSupported = (trainingMode == 0);
+    if (trainingMode != 0)
+    {
+        return;
+    }
+
+    if (context->GetInputCount() < 5)
+    {
+        return;
+    }
+
+    // Get the data type of each tensor.
+    MLOperatorEdgeDescription operatorEdgeDescription[5];
+    for (uint32_t i = 0; i < 5; ++i)
+    {
+        if (FAILED(context->GetInputEdgeDescription(i, &operatorEdgeDescription[i]))
+        ||  operatorEdgeDescription[i].edgeType != MLOperatorEdgeType::Tensor)
+        {
+            return;
+        }
+    }
+
+    // Fall back if the data types of the mean/variance or scale/bias differ from the input.
+    MLOperatorTensorDataType inputTensorDataType = operatorEdgeDescription[0].tensorDataType;
+    for (uint32_t i = 1; i < 5; ++i)
+    {
+        if (operatorEdgeDescription[i].tensorDataType != inputTensorDataType)
+        {
+            return;
+        }
+    }
+
+    *isSupported = true;
 }
 
 DML_OP_DEFINE_CREATION_FUNCTION(BatchNormalization, DmlOperatorBatchNormalization);
-DML_OP_DEFINE_CREATION_FUNCTION(BatchNormalization15, DmlOperatorBatchNormalization);
```

Comment:
I think we need to register version 15 also here, otherwise it will always pick same implementations for all opsets including 15.
Please ignore this, if I missed something here.

Reply:
Note that in OperatorRegistration.cpp, I changed it to a normal non-versioned operator. So BatchNormalization will be called now.

Reply:
As discussed offline, DmlOperatorBatchNormalization15 is not used as of now. If ospet 15 specific functionality will be used, then it will fall back to CPU EP.